tensorflow/core/ops/image_ops.cc

```diff
@@ -558,6 +558,19 @@ contents: 0-D. PNG-encoded image.
 )doc");
 
 // --------------------------------------------------------------------------
+REGISTER_OP("DecodeBmp")
+    .Input("contents: string")
+    .Output("image: uint8")
```

Comment:
Two thoughts here:

1) uint8 isn't the only possible output: if the bpp is 32 bits, we'd want to be able to produce uint32 tensors, right?  If so, perhaps we can do:

.Output("image: T")
.Attr("T: {uint8, uint16, uint32} = DT_UINT8")

I'm not sure what the default should be: are most bmp's only bpp == 8 ?

You then would need to templatize the OpKernel by output type, and register different kernels using TypeConstraint on REGISTER_KERNEL; let me know if you'd like some more pointers about how to do this!